Prevent metadata extraction phase from failing when dealing with an encrypted PDF file

Originally requested in: https://issue.openaire.research-infrastructures.eu/issues/8826.

IIS metadata extraction phase fails with the following issue:
```
2023-06-14 11:31:49,762 FATAL [IPC Server handler 27 on 38172] org.apache.hadoop.mapred.TaskAttemptListenerImpl: Task: attempt_1676554762303_49073_m_001124_3 - exited : java.lang.ClassNotFoundException: org.bouncycastle.cert.X509CertificateHolder
    at java.net.URLClassLoader.findClass(URLClassLoader.java:381)
    at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
    at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:331)
    at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
    at com.itextpdf.text.pdf.PdfReader.readDecryptedDocObj(PdfReader.java:985)
    at com.itextpdf.text.pdf.PdfReader.readDocObj(PdfReader.java:1469)
    at com.itextpdf.text.pdf.PdfReader.readPdf(PdfReader.java:751)
    at com.itextpdf.text.pdf.PdfReader.<init>(PdfReader.java:198)
    at com.itextpdf.text.pdf.PdfReader.<init>(PdfReader.java:412)
    at com.itextpdf.text.pdf.PdfReader.<init>(PdfReader.java:432)
    at pl.edu.icm.cermine.structure.ITextCharacterExtractor.extractCharacters(ITextCharacterExtractor.java:97)
    at pl.edu.icm.cermine.ExtractionUtils.extractCharacters(ExtractionUtils.java:60)
    at pl.edu.icm.cermine.InternalContentExtractor.doWork(InternalContentExtractor.java:346)
    at pl.edu.icm.cermine.InternalContentExtractor.doWork(InternalContentExtractor.java:339)
    at pl.edu.icm.cermine.InternalContentExtractor.doWork(InternalContentExtractor.java:339)
    at pl.edu.icm.cermine.InternalContentExtractor.doWork(InternalContentExtractor.java:339)
    at pl.edu.icm.cermine.InternalContentExtractor.doWork(InternalContentExtractor.java:339)
    at pl.edu.icm.cermine.InternalContentExtractor.doWork(InternalContentExtractor.java:339)
    at pl.edu.icm.cermine.InternalContentExtractor.doWork(InternalContentExtractor.java:339)
    at pl.edu.icm.cermine.InternalContentExtractor.getContentAsNLM(InternalContentExtractor.java:299)
    at pl.edu.icm.cermine.ContentExtractor.getContentAsNLM(ContentExtractor.java:662)
    at pl.edu.icm.cermine.ContentExtractor.getContentAsNLM(ContentExtractor.java:678)
    at eu.dnetlib.iis.wf.metadataextraction.MetadataExtractorMapper.handleContent(MetadataExtractorMapper.java:257)
    at eu.dnetlib.iis.wf.metadataextraction.MetadataExtractorMapper.processStream(MetadataExtractorMapper.java:236)
    at eu.dnetlib.iis.wf.metadataextraction.MetadataExtractorMapper.map(MetadataExtractorMapper.java:191)
    at eu.dnetlib.iis.wf.metadataextraction.MetadataExtractorMapper.map(MetadataExtractorMapper.java:49)
    at org.apache.hadoop.mapreduce.Mapper.run(Mapper.java:145)
    at org.apache.hadoop.mapred.MapTask.runNewMapper(MapTask.java:793)
    at org.apache.hadoop.mapred.MapTask.run(MapTask.java:341)
    at org.apache.hadoop.mapred.YarnChild$2.run(YarnChild.java:164)
    at java.security.AccessController.doPrivileged(Native Method)
    at javax.security.auth.Subject.doAs(Subject.java:422)
    at org.apache.hadoop.security.UserGroupInformation.doAs(UserGroupInformation.java:1924)
    at org.apache.hadoop.mapred.YarnChild.main(YarnChild.java:158)
```

after further inspection it turned out the following, encrypted PDF (identified with the `4d89a9b570802e5f35ff8548387cd138` checksum) caused the fatal error:

https://zaguan.unizar.es/record/85818/files/TAZ-TFG-2019-516.pdf

It is enough to add the following dependency among the IIS libs:

https://mvnrepository.com/artifact/org.bouncycastle/bcpkix-jdk15on/1.50

After running some tests in isolation it solved the problem in a way metadata extraction should not fail with fatal `ClassNotFoundException` but with already managed:

```
Exception occured: pl.edu.icm.cermine.exception.AnalysisException: Invalid PDF file
    at pl.edu.icm.cermine.structure.ITextCharacterExtractor.extractCharacters(ITextCharacterExtractor.java:122)
    at pl.edu.icm.cermine.ExtractionUtils.extractCharacters(ExtractionUtils.java:60)
    at pl.edu.icm.cermine.InternalContentExtractor.doWork(InternalContentExtractor.java:348)
    at pl.edu.icm.cermine.InternalContentExtractor.getImages(InternalContentExtractor.java:171)
    at pl.edu.icm.cermine.ContentExtractor.getImages(ContentExtractor.java:290)
    at pl.edu.icm.cermine.ContentExtractor.getImages(ContentExtractor.java:307)
    at pl.edu.icm.cermine.ContentExtractor.main(ContentExtractor.java:806)
Caused by: com.itextpdf.text.exceptions.InvalidPdfException
    at com.itextpdf.text.pdf.PdfReader.readPdf(PdfReader.java:757)
    at com.itextpdf.text.pdf.PdfReader.<init>(PdfReader.java:198)
    at com.itextpdf.text.pdf.PdfReader.<init>(PdfReader.java:412)
    at com.itextpdf.text.pdf.PdfReader.<init>(PdfReader.java:432)
    at pl.edu.icm.cermine.structure.ITextCharacterExtractor.extractCharacters(ITextCharacterExtractor.java:97)
which is not fatal and is already handled by the metadata extraction job.
```

--- a/pom.xml
+++ b/pom.xml
@@ -649,6 +649,13 @@
                 <artifactId>bcprov-jdk15on</artifactId>
                 <version>1.50</version>
             </dependency>
+            
+            <!-- #1412 fix, required when dealing with encrypted PDF files -->
+            <dependency>
+                <groupId>org.bouncycastle</groupId>
+                <artifactId>bcpkix-jdk15on</artifactId>
+                <version>1.50</version>
+            </dependency>
 
             <!-- Needed to run Avro-based map-reduce -->
             <dependency>

